Octokit/Http/CustomAwaiter.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+namespace Octokit
+{
+    public static class CustomAwaiter
+    {
+        public static TaskAwaiter<IReadOnlyList<T>> GetAwaiter<T>(this IDeferredRequest<T> request)
```

Comment:
So I could :fire: this awaiter and make the user call `.ToTask()` for these methods.

It's less magical, but takes away the hot/cold issue.